### ISSUE TITLE
Convergence failures when there are a small number of significant hits in an annotation

### ISSUE THREAD

**Issue by joepickrell** [joepickrell/fgwas#4]
If there are only a small number of GWAS hits in an annotation, fgwas sometimes fails to converge, as it "tries" to estimate the MLE of the enrichment parameter as -Inf. A simple fix is to only use the penalized likelihood (-onlyp -print). 

A more involved approach would be to implement an explicit prior on the enrichment parameters and integrate over those.

**Comment by swvanderlaan** [comment]
I've done a GWAS. And so far only have one hit. Now I want to fine map it. Would it work with this package considering the above? What is your advice. Do you know of alternatives?

**Comment by Jeremy37** [comment]
I don't think that any fine mapping / annotation enrichment methods are likely to work well when you have such a limited number of associations. If you think there are lots of sub-threshold associations you could still try. A danger is that if you test fgwas with lots of different annotations then you are likely to overfit your data, and the "best" annotation might not truly reflect enrichment of causal variants.
For fine mapping just one association, I would suggest instead looking at your credible set of SNPs (e.g. those that comprise 95% or 99% of the statistical probability of association without any annotations), and seeing what overlaps you find in HaploReg (http://archive.broadinstitute.org/mammals/haploreg/haploreg.php). (If your SNP positions are in hg37 you need to use an older version of HaploReg.)
This doesn't estimate a posterior probability using the annotation data, but such an estimate would be unreliable in an underpowered GWAS anyway.

**Comment by swvanderlaan** [comment]
Thanks for the reply. To be clear, with one hit, I meant: one locus with a leadSNP, obvious there are more variants for that locus at genome-wide significance. Does it then work? Or still not?

**Comment by Jeremy37** [comment]
Yes, one associated locus is very little data from which to estimate annotation enrichments. It might give you an answer - I just would be skeptical how reliable the answer is. You would still want to use your genome-wide data, as sub-threshold associations might contribute something, and would make the estimate more robust (the prevalence of non-associated SNPs in an annotation is important for estimating enrichment of causal SNPs in the annotation).
